src/ts-transform.ts

```diff
@@ -0,0 +1,231 @@
+import * as ts from 'typescript';
+
+abstract class Visitor {
+  protected context: ts.TransformationContext;
+
+  constructor(context: ts.TransformationContext) {
+    this.context = context;
+  }
+
+  abstract visit(node: ts.Node): ts.VisitResult<ts.Node>;
+}
+
+function stripDecorator(node: ts.Node, name: string): ts.Node {
+  if (!node.decorators) {
+    return node;
+  }
+
+  const newDecorators = node.decorators.filter(
+    (d) => getDecoratorName(d) !== name);
+
+
+  if (newDecorators.length === 0) {
+    node.decorators = undefined;
+  } else {
+    node.decorators = ts.createNodeArray(newDecorators);
+  }
+
+  return node;
+}
+
+function getDecoratorName(decorator: ts.Decorator): string {
+  if (ts.isCallExpression(decorator.expression)) {
+    return decorator.expression.expression.getText();
+  }
+  return decorator.expression.getText();
+}
+
+function findMatchingDecorators(node: ts.Node, name: string): ts.Decorator[] {
+  const decorators = node.decorators;
+
+  if (!decorators) {
+    return [];
+  }
+
+  const matchingDecorators = decorators.filter(
+      (d) => name === getDecoratorName(d));
+
+  return matchingDecorators;
+}
+
+function getDecoratorArguments(decorator: ts.Decorator): ts.Expression[] {
+  if (ts.isCallExpression(decorator.expression)) {
+    return [...decorator.expression.arguments];
+  }
+  return [];
+}
+
+interface IDecoratorVisitor {
+  kind: string;
+}
+
+class CustomElementVisitor extends Visitor implements IDecoratorVisitor {
+  public kind = 'customElement';
+
+  constructor(context: ts.TransformationContext) {
+    super(context);
+  }
+
+  visit(node: ts.Node): ts.VisitResult<ts.Node> {
+    if (!ts.isClassDeclaration(node)) {
+      return ts.visitEachChild(node, (child) =>
+        this.visit(child), this.context);
+    }
+
+    const matchingDecorators = findMatchingDecorators(node, this.kind);
+
+    if (matchingDecorators.length === 0) {
+      return node;
+    }
+
+    stripDecorator(node, this.kind);
+
+    const args = getDecoratorArguments(matchingDecorators[0]);
+
+    if (args.length === 0) {
+      return node;
+    }
+
+    const name = args[0];
+    const defineCall = ts.createCall(

```

Comment:
this is wrong at the minute. i think its either backwards or its missing the second argument!